lib/Conversion/CalyxToFSM/CalyxToFSM.cpp

```diff
@@ -268,6 +268,71 @@ LogicalResult CompileFSMVisitor::visit(StateOp currentState, EnableOp enableOp,
   return success();
 }
 
+// CompileInvoke is used to convert invoke operations to group operations and
+// enable operations.
+class CompileInvoke {
+public:
+  CompileInvoke(ComponentOp component, OpBuilder builder)
+      : component(component), builder(builder) {}
+  void compile();
+
+private:
+  void lowerInvokeOp(InvokeOp invokeOp);
+  ComponentOp component;
+  OpBuilder builder;
+};
+
+// Access all invokeOp.
+void CompileInvoke::compile() {
+  llvm::SmallVector<InvokeOp> invokeOps =
+      component.getControlOp().getInvokeOps();
+  for (InvokeOp op : invokeOps) {
+    lowerInvokeOp(op);
+  }
+}
+
+// Convert an invoke operation to a group operation and an enable operation.
+void CompileInvoke::lowerInvokeOp(InvokeOp invokeOp) {
+
+  // Create a ConstantOp to assign a value to the go port.
+  Operation *prevNode = component.getWiresOp().getOperation()->getPrevNode();
+  builder.setInsertionPointAfter(prevNode);
+  hw::ConstantOp constantOp = builder.create<hw::ConstantOp>(
+      prevNode->getLoc(), builder.getI1Type(), 1);
+  Location loc = component.getWiresOp().getLoc();
+
+  // Set the insertion point at the end of the wires block.
+  builder.setInsertionPointToEnd(component.getWiresOp().getBodyBlock());
+  llvm::StringRef callee = invokeOp.getCallee();
+  std::string groupNameHead = "invoke_" + callee.str() + "_";
+  size_t groupNameTail = 0;
+  std::string groupName = groupNameHead + std::to_string(groupNameTail);
+
+  // Check if the group name is correct.
+  while (component.getWiresOp().lookupSymbol(groupName))
+    groupName = groupNameHead + std::to_string(++groupNameTail);
```

Comment:
To avoid some code duplication: 
```suggestion

std::string groupName;
do {
  groupName = groupNameHead + std::to_string(groupNameTail++);
} while (component.getWiresOp().lookupSymbol(groupName))
```